conf/db.c

```diff
@@ -380,43 +443,53 @@ M0_INTERNAL int m0_confdb_create(struct m0_be_seg      *seg,
 M0_INTERNAL void m0_confdb_destroy_credit(struct m0_be_seg *seg,
 					  struct m0_be_tx_credit *accum)
 {
-	struct m0_be_btree *btp;
-	struct m0_be_btree  btree = { .bb_seg = seg };
-	int                 rc;
+	uint8_t              *rnode;
+	struct m0_btree       btree;
+	struct m0_btree_op    b_op  = {};
+	int                   rc;
+	struct m0_btree_type  bt    = { .tt_id = M0_BT_CONFDB,
+					.ksize = sizeof (struct m0_fid),
+					.vsize = -1,
+				      };
+	uint32_t              rnode_sz_shift;
 
 	M0_ENTRY();
 
-	rc = m0_be_seg_dict_lookup(seg, btree_name, (void **)&btp);
-	if (rc == 0) {
-		m0_be_btree_destroy_credit(btp, accum);
-	} else {
-		m0_be_btree_destroy_credit(&btree, accum);
-	}
+	rc = m0_be_seg_dict_lookup(seg, btree_name, (void **)&rnode);
+	if (rc == 0 &&
+	    M0_BTREE_OP_SYNC_WITH_RC(&b_op,
+				     m0_btree_open(rnode, rnode_sz,
+						   &btree, seg, &b_op)) == 0) {
+		m0_btree_destroy_credit(&btree, accum);
+	} else
+		/** Use the same credit count as btree_create. */
+		m0_btree_create_credit(&bt, accum);
 	m0_be_seg_dict_delete_credit(seg, btree_name, accum);
-	M0_BE_FREE_CREDIT_PTR(&btree, seg, accum);
+
+	M0_ASSERT(rnode_sz > 0 && m0_is_po2(rnode_sz));
```

Comment:
It will be better to have this assert at start of function as m0_btree_open is also using these variables.